### ISSUE TITLE
Turn bulb ON on startup

### ISSUE THREAD

**Issue by ReanuKeeves01** [openshwprojects/OpenBK7231T_App#432]
Relatively simple wish, when you take the power off the bulb, and then switch it back on. It's always in the OFF state. 

I'd like it to always start in the ON state when power OFF and ON again.

Furthermore i noticed that when this one boots, you'd have to manually set the temperature first to ANY value or it won't turn on at all. It would be nice to also have this value set to 'any temperature value' really on startup. If you take them from the power now, i always have to first access them through the WebGUI to 'reset' them so that everything works again in Ha/Homekit.

Many thanks in advance. 

I'm using this pin config, with a Tuya CW/WW 820lumen bulb:

<img width="589" alt="Screenshot 2022-11-06 at 14 00 50" src="https://user-images.githubusercontent.com/114618836/200172197-b2865a80-c5df-4aa9-a3ba-e694a51a2879.png">

**Comment by valeklubomir** [comment]
Here it is:
![image](https://user-images.githubusercontent.com/71550793/200172312-64d9c429-cf47-4d11-9fe3-c8f73fe01e8b.png)
This should do the trick.

**Comment by GravityRZ** [comment]
that's odd
i have exactly the opposite no matter what flag(12) i enable or -1 at startup.

if i leave my light in the off position, take the power off and put the power back on in always goes to the ON position
i have a BP5758D driver based light

maybe something to look into

https://github.com/openshwprojects/OpenBK7231T_App/issues/401

**Comment by valeklubomir** [comment]
Actually device should this way imitate original firmware which is always ON after power is back.
To allow classic operation by turning AC power OFF/ON withou smart App. And use SmartAPP to configure device state after ON.

**Comment by valeklubomir** [comment]
May be new feature Flag XX to Define OFF state as startup value.

**Comment by GravityRZ** [comment]
maybee i am missing the point but both flag12 as well as -1 for startup value should already do this(but dowes not work in my case)

**Comment by openshwprojects** [comment]
I will have to check your report, but I will say that there is of course a way to setup light in "short startup command" by putting there a "led_enableAll 0" etc etc commands

**Comment by GravityRZ** [comment]
the problem is not that the light startsup either ON or OFF
the problem is that is does not go back to the last state.

so if the light is ON, pull the plug and connect result: ON
if the light is OFF, pull the plug(or restart) and connect result: ON(should be off)

so flag 12 does not work for the powerstate, at least not with a BP5758D controlled light
start value -1 does not work with a BP5758D controlled light

i suspect it does work with PWM lights otherwise there would be way more posts about this

**Comment by openshwprojects** [comment]
@GravityRZ is correct. The enableAll state was just not saved.

Please wait 15 minutes.

**Comment by openshwprojects** [comment]
It should be way better now, can anyone check? Also please see new flags added.

**Comment by GravityRZ** [comment]
tested it on build 1.14.137

Flag 12 now also saves powerstate(  short blink nut light stays of when it was of)

also tested flag17
when enabled and the light is off the light will turn on when changing brightness, temperature or color

so far looking good, good work

**Comment by openshwprojects** [comment]
Thank you for testing, @GravityRZ , your input is very  valuable. I was not aware about this bug until today. But it's fixed now, so I am closing the issue.

**Comment by ReanuKeeves01** [comment]
I had indeed worked out the 'startup command' solution, which worked for me. Then realized I could also simply input the startup values for the pins directly through the interface (which largely solved my issue). Upsid was that the light would always nicely come on when the power is Switched on (Wife approval rating + ) however the downside was that the brightness/temperature is always defaulted back to that setting, regardless of any former state, which is a little annoying. 

I'm gonna play around with the new update and see what works best for me, good to see there are multiple solutions to this issue.

Many thanks for the input and great to see that my request prompted another fix and ultimately a new feature. I'm gonna update my devices today and check it out immediately. 

Thanks @openshwprojects and @GravityRZ

**Comment by jcam** [comment]
Looking back at this old issue, I was wanting a configuration where the light remembers the brightness and color or color temperature, but always turns "on", even if it was off before it lost power, and without any strange blinks at startup.
I have a BP5758D lamp.
Solution: 
set flag 12 -> checked
set startup command to led_enableAll 1

I thought that I needed the BP5758D_Map 2 1 0 3 4 to be in the startup command text, and using backlog BP5758D_Map 2 1 0 3 4; led_enableAll 1 worked, but it caused the light to flash on every startup in an annoying way.
Fortunately, I found that flag 12 *also* remembers the BP5758D_Map setting!

Figured I'd comment here in case anyone else finds themselves in the same situation
